### PR TITLE
fix: upgrade traefik client library to support new resource packaging

### PR DESCRIPTION
**Pull request by christian-vdz**
## Motivation
As it stands, Switchboard cannot work with newer versions of Traefik as describe in https://github.com/borchero/switchboard/issues/111. This change patches the project to include the newer version of the Traefik client library allowing it to operate with newer versions of traefik.

Resolves https://github.com/borchero/switchboard/issues/111

## Explanation
This change upgrades the Traefik client library, patches the various files that use the Traefik, and finally updates the ClusterRole defined in the helm chart to support watching the appropriate resources.

## Changes
- Patches go.mod
- Updates library usage
- Updates the kubernetes controller permissions.

## Checklist

- [x] I added relevant tests
- [x] I have updated the documentation if necessary

### CODE REVIEW

**Comment by christian-vdz** [comment]
@borchero can you approve the workflow run?

File: justfile [hunk 1]
```diff
@@ -71,10 +71,10 @@ setup-cluster: create-cluster
   kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.13.9/config/manifests/metallb-native.yaml
   kubectl wait -n metallb-system --for=condition=ready pod -l app=metallb --timeout=240s
   kubectl apply -f https://kind.sigs.k8s.io/examples/loadbalancer/metallb-config.yaml
-  kubectl apply -f https://raw.githubusercontent.com/traefik/traefik/v2.6/docs/content/reference/dynamic-configuration/traefik.containo.us_ingressroutes.yaml
+  kubectl apply -f https://raw.githubusercontent.com/traefik/traefik/v2.10/docs/content/reference/dynamic-configuration/traefik.io_ingressroutes.yaml
   kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.11.0/cert-manager.yaml
   helm repo add bitnami https://charts.bitnami.com/bitnami
-  helm upgrade --install --set crd.create=true --wait external-dns bitnami/external-dns
+  helm upgrade --install --set crd.create=true --wait external-dns bitnami/external-dns --debug --timeout 10m
```
Comment (borchero) [inline justfile:77]: Are these two flags necessary?

```suggestion
  helm upgrade --install --set crd.create=true --wait external-dns bitnami/external-dns
```

**Review by borchero — COMMENTED** [review]
Nice, CI passes 😍 do you know what exactly was the fix?

**Comment by christian-vdz** [comment]
> Nice, CI passes 😍 do you know what exactly was the fix?

I don’t understand why it works now, I did not change related to that CI job since last week.

File: justfile (hunk 1 above)
Comment (christian-vdz) [reply justfile:77]: We can remove the debug flag, but I think that the timeout option will be required for this to work

**Comment by christian-vdz** [comment]
Hello @borchero, now that the CI passes, can you take the lead on merging this MR and creating a new release?
I'd be happy to test it on my lab ASAP! ;)

**Review by borchero — APPROVED** [review]
Thanks for the effort @christian-vdz, this looks good to me! I don't think we need to continue supporting `traffic.containo.us` (as discussed in #111) since this repo does not see too many changes :eyes:

Sorry about the long review period, I am not working with Traefik/Kubernetes at the moment and I briefly needed to catch up what exactly changed in Traefik that necessitates this change.

**Comment by borchero** [comment]
I'm briefly taking care of all the dependabot PRs and will publish a new release `0.6.0` tonight :)

**Comment by borchero** [comment]
Here we go: https://github.com/borchero/switchboard/releases/tag/v0.6.0 :smile: sorry again for the suboptimal maintenance recently.

**Comment by christian-vdz** [comment]
> Here we go: https://github.com/borchero/switchboard/releases/tag/v0.6.0 😄 sorry again for the suboptimal maintenance recently.

No worries, thank you for taking the time!  
I've tested it and it works 🥳

**Pull request merged**